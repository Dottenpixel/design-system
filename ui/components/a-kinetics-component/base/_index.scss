// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

/**
 * @summary Create a-kinetics-component component
 *
 * @name base
 * @selector .slds-a-kinetics-component
 * @restrict span
 * @support dev-ready
 * @variant
 */
.slds-a-kinetics-component {
  background-color: $color-background-dark;
  padding: $spacing-xx-small $spacing-x-small;
  border-radius: $border-radius-pill;
  font-size: $font-size-2;
  font-weight: $font-weight-bold;
  line-height: normal;
  color: $color-text-default;
  white-space: nowrap;
  display: inline-flex;
  align-items: center;

  + .slds-a-kinetics-component {
    margin-left: $spacing-x-small;
  }

  &:empty {
    padding: 0;
  }
}

/**
 * @summary Change a-kinetics-component color to a dark with light text
 * @name inverse
 * @selector .slds-a-kinetics-component_inverse
 * @restrict .slds-a-kinetics-component
 * @modifier
 * @group color
 */
.slds-a-kinetics-component_inverse {
  background-color: $a-kinetics-component-color-background-inverse;
  color: $color-text-inverse;
}

/**
 * @summary Change a-kinetics-component color to white with normal text
 * @name lightest
 * @selector .slds-a-kinetics-component_lightest
 * @restrict .slds-a-kinetics-component
 * @modifier
 * @group color
 */
.slds-a-kinetics-component_lightest {
  border: $border-width-thin solid $color-border;
  background-color: $a-kinetics-component-color-background-lightest;
  font-size: $font-size-2;
  text-transform: none;
  letter-spacing: normal;
}

/**
 * @summary Adds an icon to a-kinetics-component
 * @name icon-container
 * @selector .slds-a-kinetics-component__icon
 * @restrict .slds-a-kinetics-component span
 * @modifier
 */
.slds-a-kinetics-component__icon {
  color: $color-text-icon-default;
  vertical-align: middle;
  display: inline-flex;
  align-items: center;
  line-height: 1;

  /**
   * @summary Change a-kinetics-component icon fill color to inverse
   * @name icon-inverse
   * @selector .slds-a-kinetics-component__icon_inverse
   * @restrict .slds-a-kinetics-component__icon
   * @modifier
   * @group color
   */
  &.slds-a-kinetics-component__icon_inverse {
    color: currentColor;
  }
}

/**
 * @summary Adds an icon to the left of the a-kinetics-component text
 * @name icon-left
 * @selector .slds-a-kinetics-component__icon_left
 * @restrict .slds-a-kinetics-component__icon
 * @modifier
 * @group spacing
 */
.slds-a-kinetics-component__icon_left {
  margin-right: $spacing-xx-small;
}

/**
 * @summary Adds an icon to the right of the a-kinetics-component text
 * @name icon-right
 * @selector .slds-a-kinetics-component__icon_right
 * @restrict .slds-a-kinetics-component__icon
 * @modifier
 * @group spacing
 */
.slds-a-kinetics-component__icon_right {
  margin-left: $spacing-xx-small;
}
