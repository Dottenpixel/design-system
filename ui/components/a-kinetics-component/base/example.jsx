// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import {
  AKineticsComponent,
  InverseAKineticsComponent,
  LightestAKineticsComponent,
  AKineticsComponentIcon
} from '..';

export default (
  <AKineticsComponent>AKineticsComponent Label</AKineticsComponent>
);

export let examples = [
  {
    id: 'inverse',
    label: 'Inverse',
    element: (
      <InverseAKineticsComponent>
        Inverse a-kinetics-component
      </InverseAKineticsComponent>
    )
  },
  {
    id: 'with-left-icon',
    label: 'With left icon',
    element: (
      <AKineticsComponent>
        <AKineticsComponentIcon symbol="moneybag" />
        423 Credits Available
      </AKineticsComponent>
    )
  },
  {
    id: 'with-right-icon',
    label: 'With right icon',
    element: (
      <AKineticsComponent>
        423 Credits Available
        <AKineticsComponentIcon symbol="moneybag" align="right" />
      </AKineticsComponent>
    )
  },
  {
    id: 'light-with-left-icon',
    label: 'Light with left icon',
    element: (
      <LightestAKineticsComponent>
        <AKineticsComponentIcon symbol="moneybag" />
        423 Credits Available
      </LightestAKineticsComponent>
    )
  },
  {
    id: 'light-with-right-icon',
    label: 'Light with right icon',
    element: (
      <LightestAKineticsComponent>
        423 Credits Available
        <AKineticsComponentIcon symbol="moneybag" align="right" />
      </LightestAKineticsComponent>
    )
  },
  {
    id: 'inverse-with-left-icon',
    label: 'Inverse with left icon',
    element: (
      <InverseAKineticsComponent>
        <AKineticsComponentIcon symbol="moneybag" isInverse />
        423 Credits Available
      </InverseAKineticsComponent>
    )
  },
  {
    id: 'inverse-with-right-icon',
    label: 'Inverse with right icon',
    element: (
      <InverseAKineticsComponent>
        423 Credits Available
        <AKineticsComponentIcon symbol="moneybag" align="right" isInverse />
      </InverseAKineticsComponent>
    )
  },
  {
    id: 'a-kinetics-component-with-child',
    label: 'With Child Element',
    element: (
      <AKineticsComponent>
        Component In: <em>Lightning</em>
      </AKineticsComponent>
    )
  }
];

export let states = [
  {
    id: 'success',
    label: 'Success',
    element: (
      <AKineticsComponent isSuccess>
        Success a-kinetics-component
      </AKineticsComponent>
    )
  },
  {
    id: 'warning',
    label: 'Warning',
    element: (
      <AKineticsComponent isWarning>
        Warning a-kinetics-component
      </AKineticsComponent>
    )
  },
  {
    id: 'error',
    label: 'Error',
    element: (
      <AKineticsComponent isError>
        Error a-kinetics-component
      </AKineticsComponent>
    )
  }
];
