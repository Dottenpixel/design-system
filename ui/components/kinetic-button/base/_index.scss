// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

/**
 * @summary Create kinetic-button component
 *
 * @name base
 * @selector .slds-kinetic-button
 * @restrict span
 * @support dev-ready
 * @variant
 */
.slds-kinetic-button {
  background-color: $color-background-dark;
  padding: $spacing-xx-small $spacing-x-small;
  border-radius: $border-radius-pill;
  font-size: $font-size-2;
  font-weight: $font-weight-bold;
  line-height: normal;
  color: $color-text-default;
  white-space: nowrap;
  display: inline-flex;
  align-items: center;

  + .slds-kinetic-button {
    margin-left: $spacing-x-small;
  }

  &:empty {
    padding: 0;
  }
}

/**
 * @summary Change kinetic-button color to a dark with light text
 * @name inverse
 * @selector .slds-kinetic-button_inverse
 * @restrict .slds-kinetic-button
 * @modifier
 * @group color
 */
.slds-kinetic-button_inverse {
  background-color: $kinetic-button-color-background-inverse;
  color: $color-text-inverse;
}

/**
 * @summary Change kinetic-button color to white with normal text
 * @name lightest
 * @selector .slds-kinetic-button_lightest
 * @restrict .slds-kinetic-button
 * @modifier
 * @group color
 */
.slds-kinetic-button_lightest {
  border: $border-width-thin solid $color-border;
  background-color: $kinetic-button-color-background-lightest;
  font-size: $font-size-2;
  text-transform: none;
  letter-spacing: normal;
}

/**
 * @summary Adds an icon to kinetic-button
 * @name icon-container
 * @selector .slds-kinetic-button__icon
 * @restrict .slds-kinetic-button span
 * @modifier
 */
.slds-kinetic-button__icon {
  color: $color-text-icon-default;
  vertical-align: middle;
  display: inline-flex;
  align-items: center;
  line-height: 1;

  /**
   * @summary Change kinetic-button icon fill color to inverse
   * @name icon-inverse
   * @selector .slds-kinetic-button__icon_inverse
   * @restrict .slds-kinetic-button__icon
   * @modifier
   * @group color
   */
  &.slds-kinetic-button__icon_inverse {
    color: currentColor;
  }
}

/**
 * @summary Adds an icon to the left of the kinetic-button text
 * @name icon-left
 * @selector .slds-kinetic-button__icon_left
 * @restrict .slds-kinetic-button__icon
 * @modifier
 * @group spacing
 */
.slds-kinetic-button__icon_left {
  margin-right: $spacing-xx-small;
}

/**
 * @summary Adds an icon to the right of the kinetic-button text
 * @name icon-right
 * @selector .slds-kinetic-button__icon_right
 * @restrict .slds-kinetic-button__icon
 * @modifier
 * @group spacing
 */
.slds-kinetic-button__icon_right {
  margin-left: $spacing-xx-small;
}
