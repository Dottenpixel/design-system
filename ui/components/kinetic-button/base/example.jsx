// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import { Button } from '../../buttons/base/example';

export default <div>Nothing to see here</div>;

export let examples = [
  {
    id: 'kinetic-button',
    label: 'Kinetic Button',
    demoStyles: 'text-align: center;',
    element: (
      <Button isNeutral isKinetic>
        <svg class="slds-kinetic-border-swirl">
          <defs>
            <linearGradient
              id="slds-button_kinetic-gradient"
              x1="0%"
              y1="75%"
              x2="100%"
              y2="100%"
            >
              <stop class="slds-kinetic-gradient__stop-1" offset="0%" />
              <stop class="slds-kinetic-gradient__stop-2" offset="100%" />
            </linearGradient>
          </defs>
          <rect
            class="slds-kinetic-border-swirl-shape"
            x="0"
            y="0"
            rx="4"
            ry="4"
            fill="none"
            width="100%"
            height="100%"
          />
        </svg>
        <span class="slds-button_kinetic_label">I have KINETICS!</span>
      </Button>
    )
  }
];
