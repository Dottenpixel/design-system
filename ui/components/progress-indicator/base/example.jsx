// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React, { Fragment } from 'react';
import classNames from 'classnames';
import ButtonIcon from '../../button-icons/';
import { Tooltip } from '../../tooltips/base/example';
import {
  Modal,
  ModalHeader,
  ModalContent,
  ModalFooter
} from '../../modals/base/example';
import { ProgressBar } from '../../progress-bar';
import { KineticsBorderSwirl } from '../../kinetic-button/index';

/// ///////////////////////////////////////////
// Partial(s)
/// ///////////////////////////////////////////

export let Progress = ({
  className,
  children,
  active,
  done,
  error,
  ...props
}) => (
  <div className={classNames('slds-progress', className)}>
    <ol className="slds-progress__list">{children}</ol>
    <ProgressBar className="slds-progress-bar_x-small" value={props.value} />
  </div>
);

export let Step = ({
  className,
  children,
  active,
  done,
  error,
  isKinetic,
  ...props
}) => (
  <li
    className={classNames(
      'slds-progress__item',
      className,
      isKinetic &&
        'slds-kinetic-progress__item slds-kinetic-progress__item_activated',
      active && 'slds-is-active',
      done && 'slds-is-completed',
      error && 'slds-has-error'
    )}
  >
    {done && !error ? (
      <ButtonIcon
        className="slds-progress__marker slds-progress__marker_icon"
        symbol="success"
        aria-describedby={props['aria-describedby']}
        assistiveText={done ? children + ' - Completed' : null}
        title={done ? children + ' - Completed' : null}
      />
    ) : error ? (
      <ButtonIcon
        className="slds-progress__marker slds-progress__marker_icon"
        symbol="error"
        aria-describedby={props['aria-describedby']}
        assistiveText={error ? children + ' - Error' : null}
        title={error ? children + ' - Error' : null}
      />
    ) : (
      <Fragment>
        {isKinetic && (
          <KineticsBorderSwirl gradientId="slds-kinetic-progress-stop-gradient" />
        )}
        <button
          className="slds-button slds-progress__marker"
          aria-describedby={props['aria-describedby']}
        >
          <span className="slds-assistive-text">
            {children} {active ? '- Active' : null}
          </span>
        </button>
      </Fragment>
    )}
  </li>
);

/// ///////////////////////////////////////////
// Export
/// ///////////////////////////////////////////

export default (
  <div className="demo-only" style={{ padding: '1rem' }}>
    <Progress value="0">
      <Step active>Step 1</Step>
      <Step>Step 2</Step>
      <Step>Step 3</Step>
      <Step>Step 4</Step>
      <Step>Step 5</Step>
    </Progress>
  </div>
);

export let states = [
  {
    id: 'next-step',
    label: 'Next Step',
    element: (
      <div className="demo-only" style={{ padding: '1rem' }}>
        <Progress value="25">
          <Step done>Step 1</Step>
          <Step active>Step 2</Step>
          <Step>Step 3</Step>
          <Step>Step 4</Step>
          <Step>Step 5</Step>
        </Progress>
      </div>
    )
  },
  {
    id: 'has-error',
    label: 'Step - Error',
    element: (
      <div className="demo-only" style={{ padding: '1rem' }}>
        <Progress value="25">
          <Step done>Step 1</Step>
          <Step error>Step 2</Step>
          <Step>Step 3</Step>
          <Step>Step 4</Step>
          <Step>Step 5</Step>
        </Progress>
      </div>
    )
  },
  {
    id: 'tooltip',
    label: 'Tooltip',
    element: (
      <div className="demo-only" style={{ padding: '3rem 1rem 0' }}>
        <Progress value="50">
          <Step done>Step 1</Step>
          <Step done>Step 2</Step>
          <Step active aria-describedby="step-3-tooltip">
            Step 3
          </Step>
          <Step>Step 4</Step>
          <Step>Step 5</Step>
        </Progress>
        <Tooltip
          className="slds-nubbin_bottom"
          id="step-3-tooltip"
          style={{
            position: 'absolute',
            top: '0px',
            left: 'calc(50% + 7px)',
            transform: 'translateX(-50%)'
          }}
        >
          Verify Email
        </Tooltip>
      </div>
    )
  }
];

export let examples = [
  {
    id: 'modal',
    label: 'In a modal',
    element: (
      <div className="demo-only" style={{ height: '640px' }}>
        <Modal className="slds-modal_large" aria-labelledby="header43">
          <ModalHeader>
            <h2 id="header43" className="slds-text-heading_medium">
              Modal Header
            </h2>
          </ModalHeader>
          <ModalContent className="slds-grow slds-p-around_medium" />
          <ModalFooter className="slds-grid slds-grid_align-spread">
            <button className="slds-button slds-button_neutral">Cancel</button>
            <Progress className="slds-progress_shade" value="25">
              <Step done>Step 1</Step>
              <Step active>Step 2</Step>
              <Step>Step 3</Step>
              <Step>Step 4</Step>
              <Step>Step 5</Step>
            </Progress>
            <button className="slds-button slds-button_brand">Save</button>
          </ModalFooter>
        </Modal>
        <div className="slds-backdrop slds-backdrop_open" />
      </div>
    )
  },
  {
    id: 'shade',
    label: 'On a gray background',
    element: (
      <div
        className="demo-only"
        style={{ background: '#F3F2F2', padding: '1rem' }}
      >
        <Progress className="slds-progress_shade" value="25">
          <Step done>Step 1</Step>
          <Step active>Step 2</Step>
          <Step>Step 3</Step>
          <Step>Step 4</Step>
          <Step>Step 5</Step>
        </Progress>
      </div>
    )
  }
];
