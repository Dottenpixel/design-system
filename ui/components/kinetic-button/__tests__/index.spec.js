/* eslint-env jest */
import React from 'react';
import {
  LightestKineticButton,
  InverseKineticButton,
  KineticButton,
  KineticButtonIcon
} from '../';
import { UtilityIcon } from '../../icons/base/example';
import createHelpers from '../../../../jest.helpers';
const { matchesMarkup } = createHelpers(__dirname);

describe('KineticButton colors', () => {
  it('renders a default kinetic-button', () =>
    matchesMarkup(<KineticButton>Lovely kinetic-button</KineticButton>));

  it('renders an inverse kinetic-button', () =>
    matchesMarkup(
      <InverseKineticButton>Lovely kinetic-button</InverseKineticButton>
    ));

  it('renders kinetic-button', () =>
    matchesMarkup(<LightestKineticButton>Light</LightestKineticButton>));
});

describe('KineticButton with icons', () => {
  it('renders kinetic-button with left icon', () =>
    matchesMarkup(
      <LightestKineticButton>
        <KineticButtonIcon symbol="world" />
        Icon on the left
      </LightestKineticButton>
    ));

  it('renders kinetic-button with right icon', () =>
    matchesMarkup(
      <LightestKineticButton>
        Icon on the right
        <KineticButtonIcon symbol="world" />
      </LightestKineticButton>
    ));

  it('renders kinetic-button with icon only', () =>
    matchesMarkup(
      <LightestKineticButton>
        <KineticButtonIcon
          symbol="world"
          assistiveText="assistive text goes here, do it!"
        />
      </LightestKineticButton>
    ));

  it('renders deprecated kinetic-button with right icon', () =>
    matchesMarkup(
      <LightestKineticButton>
        Icon on the right
        <UtilityIcon
          containerClassName="slds-m-left_xx-small"
          className="slds-icon_xx-small slds-icon-text-default"
          assistiveText={false}
          symbol="world"
        />
      </LightestKineticButton>
    ));

  it('renders deprecated kinetic-button with icon only', () =>
    matchesMarkup(
      <LightestKineticButton>
        <UtilityIcon
          className="slds-icon_xx-small slds-icon-text-default"
          assistiveText="assistive text goes here, do it!"
          symbol="world"
        />
      </LightestKineticButton>
    ));
});
