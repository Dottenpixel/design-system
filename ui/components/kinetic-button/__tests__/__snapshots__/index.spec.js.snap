// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KineticButton colors renders a default kinetic-button 1`] = `
<span
  className="slds-button_kinetic"
>
  Lovely kinetic-button
</span>
`;

exports[`KineticButton colors renders an inverse kinetic-button 1`] = `
<span
  className="slds-button_kinetic slds-button_kinetic_inverse"
>
  Lovely kinetic-button
</span>
`;

exports[`KineticButton colors renders kinetic-button 1`] = `
<span
  className="slds-button_kinetic slds-button_kinetic_lightest"
>
  Light
</span>
`;

exports[`KineticButton with icons renders kinetic-button with icon only 1`] = `
<span
  className="slds-button_kinetic slds-button_kinetic_lightest"
>
  <span
    className="slds-button_kinetic__icon slds-button_kinetic__icon_left"
  >
    <span
      className="slds-icon_container slds-icon-utility-world slds-current-color"
      title="Description of icon when needed"
    >
      <svg
        aria-hidden={true}
        className="slds-icon slds-icon_xx-small"
      >
        <use
          xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#world"
        />
      </svg>
      <span
        className="slds-assistive-text"
      >
        assistive text goes here, do it!
      </span>
    </span>
  </span>
</span>
`;

exports[`KineticButton with icons renders kinetic-button with left icon 1`] = `
<span
  className="slds-button_kinetic slds-button_kinetic_lightest"
>
  <span
    className="slds-button_kinetic__icon slds-button_kinetic__icon_left"
  >
    <span
      className="slds-icon_container slds-icon-utility-world slds-current-color"
      title="Description of icon when needed"
    >
      <svg
        aria-hidden={true}
        className="slds-icon slds-icon_xx-small"
      >
        <use
          xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#world"
        />
      </svg>
    </span>
  </span>
  Icon on the left
</span>
`;

exports[`KineticButton with icons renders kinetic-button with right icon 1`] = `
<span
  className="slds-button_kinetic slds-button_kinetic_lightest"
>
  Icon on the right
  <span
    className="slds-button_kinetic__icon slds-button_kinetic__icon_left"
  >
    <span
      className="slds-icon_container slds-icon-utility-world slds-current-color"
      title="Description of icon when needed"
    >
      <svg
        aria-hidden={true}
        className="slds-icon slds-icon_xx-small"
      >
        <use
          xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#world"
        />
      </svg>
    </span>
  </span>
</span>
`;

exports[`KineticButton with icons renders deprecated kinetic-button with icon only 1`] = `
<span
  className="slds-button_kinetic slds-button_kinetic_lightest"
>
  <span
    className="slds-icon_container slds-icon-utility-world"
    title="Description of icon when needed"
  >
    <svg
      aria-hidden={true}
      className="slds-icon slds-icon_xx-small slds-icon-text-default"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#world"
      />
    </svg>
    <span
      className="slds-assistive-text"
    >
      assistive text goes here, do it!
    </span>
  </span>
</span>
`;

exports[`KineticButton with icons renders deprecated kinetic-button with right icon 1`] = `
<span
  className="slds-button_kinetic slds-button_kinetic_lightest"
>
  Icon on the right
  <span
    className="slds-icon_container slds-icon-utility-world slds-m-left_xx-small"
    title="Description of icon when needed"
  >
    <svg
      aria-hidden={true}
      className="slds-icon slds-icon_xx-small slds-icon-text-default"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#world"
      />
    </svg>
  </span>
</span>
`;
